<?xml version="1.0"?>
<launch>
    <node name="face_detection_node" pkg="opencv_apps" type="face_detection" required="true">
        <remap from="image" to="/usb_cam/image_raw" />
        <param name="use_camera_info" value="false" />
        <param name="debug_view" value="false" />
        <param name="queue_size" value="3" />
        <param name="face_cascade_name" value="/usr/share/opencv4/haarcascades/haarcascade_frontalface_alt.xml" />
        <param name="eyes_cascade_name" value="/usr/share/opencv4/haarcascades/haarcascade_eye_tree_eyeglasses.xml" />
    </node>
    <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" required="true">
        <param name="video_device" value="/dev/video0"/>
        <param name="image_width" value="320"/>
        <param name="image_height" value="240"/>
        <param name="pixel_format" value="yuyv"/>
        <param name="framerate" value="30"/>
    </node>
    <include file="$(find wr_pi_extend_board_driver)/launch/core.launch" />
    <node pkg="face_tracking_ros" type="face_tracking_node.py" name="face_tracking_node" output="screen" required="true"/>
</launch>
